embassy-rp/src/gpio.rs

```diff
@@ -139,6 +138,91 @@ impl<'d, T: Pin> Drop for Output<'d, T> {
     }
 }
 
+/// GPIO output open-drain.
+pub struct OutputOpenDrain<'d, T: Pin> {
+    pin: T,
+    phantom: PhantomData<&'d mut T>,
+}
+
+impl<'d, T: Pin> OutputOpenDrain<'d, T> {
+    #[inline]
+    pub fn new(pin: impl Unborrow<Target = T> + 'd, initial_output: Level) -> Self {
+        unborrow!(pin);
+
+        unsafe {
+            match initial_output {
+                Level::High => {
+                    // For Open Drain high should pull low (drain).
+                    pin.pad_ctrl().write(|w| {
+                        w.set_pde(true);
+                    });
+                }
+                Level::Low => {
+                    // For Open Drain low should float the output pin (open).
+                    pin.pad_ctrl().write(|w| {
+                        w.set_od(true);
+                    });
+                }
+            }
+        }
+
+        Self {
+            pin,
+            phantom: PhantomData,
+        }
+    }
+
+    /// Set the output as high.
+    #[inline]
+    pub fn set_high(&mut self) {
+        // For Open Drain high should pull low (drain).
+        unsafe {
+            self.pin.pad_ctrl().write(|w| {
+                w.set_od(false);
```

Comment:
you can use `gpio_oe` in `SIO` to enable/disable the output  (`SIO` is faster than all other peripherals, accesses to it are single-cycle)

Reply:
That makes sense, I'll update shortly. Thanks